Mobile Buy SDK/Mobile Buy SDK/Product View/BUYProductViewController.m

```diff
@@ -444,12 +444,13 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView
 				[self.productView.productViewHeader.productViewHeaderOverlay scrollViewDidScroll:scrollView withNavigationBarHeight:CGRectGetHeight(self.navigationBar.bounds)];
 			}
 		} else if (self.productView.productViewHeader == nil && self.navigationBar && self.navigationBar.alpha == 0) {
+			// If the product view does not contain an image, inset the productView so its not obstructed with the navigation bar
 			[(BUYNavigationController*)self.navigationController updateCloseButtonImageWithTintColor:YES duration:0];
 			self.navigationBar.alpha = 1;
 			self.navigationBarTitle.alpha = 1;
 			CGFloat topInset = 0;
 			if (self.presentingViewController) {
-				topInset = CGRectGetHeight([[(UINavigationController*)self.presentingViewController navigationBar] bounds]) + self.presentingViewController.topLayoutGuide.length;
+				topInset = CGRectGetHeight(self.navigationController.navigationBar.bounds) + self.presentingViewController.topLayoutGuide.length;
```

Comment:
I'm confused, wouldn't the topLayoutGuide already represent the navigation bar's height?